### PR TITLE
fix(wait stage): Make wait stage restartable

### PR DESCRIPTION
**Pull request by marchello2000**
Noticed that when I restarted a WaitStage it didn't actually wait.
That's because it caches the stage `startTime`.
Not sure why it does since the stage should have actual start time the time the task runs, so making the wait task use that instead of caching it's own time.

### CODE REVIEW

**Comment by marchello2000** [comment]
I don't know if there was some special historical reason why this was necessary so if there was let me know and I can workaround it/abandon/reword this change - it's clearly not VERY frequent use case seeing as restarts are an admin EP

File: orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/WaitStage.java [hunk 1]
```diff
@@ -33,44 +33,33 @@
   public static String STAGE_TYPE = "wait";
 
   @Override
-  public void taskGraph(Stage stage, TaskNode.Builder builder) {
+  public void taskGraph(@NotNull Stage stage, TaskNode.Builder builder) {
```
Comment (ezimanyi) [inline orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/WaitStage.java]: The super method uses `@Nonnull` from `javax.annotation`, which is in general the non-null annotation we use more commonly (including elsewhere in this class) so I'd suggest using that here.

File: orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/tasks/WaitTask.java [hunk 2]
```diff
@@ -44,17 +41,14 @@ public WaitTask(Clock clock) {
...
     WaitStage.WaitStageContext context = stage.mapTo(WaitStage.WaitStageContext.class);
 
-    if (context.getWaitTime() == null) {
-      return TaskResult.SUCCEEDED;
-    }
-
     Instant now = clock.instant();
 
-    if (context.isSkipRemainingWait()) {
+    if (context.isSkipRemainingWait() || context.getWaitDuration().isZero()) {
```
Comment (ezimanyi) [inline orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/tasks/WaitTask.java]: This is maybe a silly minor comment, but do we need to special case the zero wait time here?  Wouldn't the start time plus zero always be before now in the next block which would simplify the number of cases to consider?

File: orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/WaitStage.java (hunk 1 above)
Comment (marchello2000) [reply orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/WaitStage.java]: silly me for blindly accepting intelliJ offer.. Thanks for catching it!

File: orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/tasks/WaitTask.java (hunk 2 above)
Comment (marchello2000) [reply orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/tasks/WaitTask.java]: yeah, good call - thank you!

Comment (marchello2000) [reply orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/tasks/WaitTask.java]: Also realized the `context.getWaitDuration()` check for `null` on line 72 wasn't necessary and removed it

**Pull request merged**